dev/com.ibm.ws.cdi.1.2.weld/src/com/ibm/ws/cdi/proxy/ProxyServicesImpl.java

```diff
@@ -53,6 +54,14 @@ public void cleanup() {
 
     @Override
     public ClassLoader getClassLoader(final Class<?> proxiedBeanType) {
+
+        if (System.getSecurityManager() != null) {
+            String className = Thread.currentThread().getStackTrace()[1].getClassName();
+            if (! className.equals("org.jboss.weld.bean.proxy.ProxyFactory")) {
+                throw new AccessControlException("Someone other than weld attempted to access proxy services");
+            }
```

Comment:
Do we need either a similar check or the removal of `doPrivileged` further down in `loadBeanClass`?

Also, it looks like `loadBeanClass` calls `getClassLoader` and that would break your current stack-walking logic since there'd be an extra call in the stack. Are we testing this code path?